### PR TITLE
Add runtime APIs to call module init, start, stop and function invocations from a Ballerina project

### LINKED ISSUE #42128: [New Feature]: Runtime API to start the new Runtime and call ballerina function
### Description

I have a ballerina program. I will compile it and generate a jar file. Then I will add that jar as dependency to another Java Project. Then I need to start the ballerina program from the external Java Project and call the functions.

### Describe your problem(s)

_No response_

### Describe your solution(s)

Provide an Runtime API as below.

```Java
import io.ballerina.runtime.api.Runtime;

Runtime balRuntime = Runtime.start(“org”, “module”, 1.0.0);
Object[] args = [];
balRuntime.invokeMethodAsync(“myFunction”, args, callback);
Runtime.stop(“org”, “module”, 1.0.0);

```

Runtime Class

```Java
Class initClass;

public static void start(String orgName, String moduleName, String version) {
String initClassName = orgNAme + “$” + moduleName + “$” + version
	initClass = Class.load(initClassName);
Scheduler s = new Scheduler();
s.schedule(moduleInitFunction);
	initClass.invoke(“moduleInit”); 
}


public static void stop(String orgName, String moduleName, String version) {
	initClass.invoke(“moduleStop”);
}
```

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #42128

## Samples
```ballerina
Runtime balRuntime = Runtime.from(new Module("orgName", "moduleName", "majorVersion"));
balRuntime.init();
balRuntime.start();
Object[] args = [];
balRuntime.invokeMethodAsync(“functionName”, args, callback);
balRuntime.stop(“org”, “module”, 1.0.0);
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java [hunk 1]
```diff
@@ -245,6 +248,66 @@ private static boolean isInvalidBallerinaValue(Object value) {
...
 
+    public static String formatErrorMessage(Throwable e) {
+        try {
+            if (e instanceof BError) {
+                return ((BError) e).getPrintableStackTrace();
+            } else if (e instanceof Exception | e instanceof Error) {
+                return getPrintableStackTrace(e);
+            } else {
+                return getPrintableStackTrace(e);
+            }
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java:259]:
```suggestion
            if (e instanceof BError error) {
                return error.getPrintableStackTrace();
            }
            if (e instanceof Exception | e instanceof Error) {
                return getPrintableStackTrace(e);
            } 
            return getPrintableStackTrace(e);
```

**Comment by warunalakshitha** [comment]
@gabilang Shall we create an issue to investigate why code coverage is low even tests are covered?

**Comment by gabilang** [comment]
Will fix the code-coverage issue separately. Created a task for it https://github.com/ballerina-platform/ballerina-lang/issues/42427

**Review by warunalakshitha — APPROVED** [review]
LGTM

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java [hunk 2]
```diff
@@ -152,4 +179,13 @@ public abstract Object invokeMethodAsync(BObject object, String methodName, Stri
...
     public abstract void registerStopHandler(BFunctionPointer<?, ?> stopHandler);
+
+    /**
+     * Invoke a Ballerina function pointer asynchronously.
+     *
+     * @param functionName  Name of the function which needs to be invoked.
+     * @param args          Arguments of the Ballerina function.
+     * @param callback      Callback which will get notified once the function execution is done.
+     */
+    public abstract void invokeMethodAsync(String functionName, Object[] args, Callback callback);
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java:190]:
```suggestion
    public abstract void invokeMethodAsync(String functionName, Callback callback, Object... args);
```

Comment (warunalakshitha) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java:190]: Shall we change the method as below to comply with other invokeMethodAsync methods

Comment (gabilang) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java]: Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/42363/commits/85939a40a21405befa32e18dcf654fa68f849c49

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**